### PR TITLE
add title on create and edit form

### LINKED ISSUE #1033: Provide title to `Create` (reference) form to remind what is being created
We have custom forms to `Create` many resource types - these include a title to explain what is being created.

We don't have one for the `Reference` data-types.

We should pass a title to the form, to explain what is being created.  Note: for the `cat-xxx` types, we need to take the user-friendly name from the `config` data.

image

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
Fixes #1033

### CODE REVIEW

File: src/resources/reference-data/ReferenceDataForm.tsx [hunk 1]
```diff
@@ -214,18 +214,29 @@ const FormContent = ({
...
+      : name === constants.R_PROTECTIVE_MARKING
+      ? 'Protective Marking'
+      : name === constants.R_MEDIA_TYPE
+      ? 'Media type'
+      : name === constants.R_VAULT
+      ? 'Vault'
       : 'resource'
   const warningTextForId = `Warning: Editing the id of a ${resourceName} that is in use may lead to data corruption.  The id of a ${resourceName} must not be modified if data has been assigned to that ${resourceName}.`
-
+  const nameOfRescource = resourceName
```
Comment (IanMayo) [inline src/resources/reference-data/ReferenceDataForm.tsx:234]: Hello @TahaKhanAbdalli - do we need to introduce a fresh `const` here?

**Comment by IanMayo** [comment]
Thanks @TahaKhanAbdalli - I think I've just found an instance of our `catXXX` values leaking into the UI:
image


The show page for `catCode`, `catCave` and `catHandle` doesn't use the name from the `configData`

**Comment by TahaKhanAbdalli** [comment]
> Thanks @TahaKhanAbdalli - I think I've just found an instance of our `catXXX` values leaking into the UI: image
> 
> The show page for `catCode`, `catCave` and `catHandle` doesn't use the name from the `configData`

Okay let me check

File: src/resources/reference-data/ReferenceDataForm.tsx [hunk 2]
```diff
@@ -212,30 +212,27 @@ const FormContent = ({
   }, [{ ...dirtyFields }])
   const validName = name ?? ''
   const configData = useConfigData()
```
Comment (IanMayo) [inline src/resources/reference-data/ReferenceDataForm.tsx:214]: Thanks @TahaKhanAbdalli - this is a much neater data structure.

File: src/resources/reference-data/ReferenceDataShow.tsx [hunk 3]
```diff
@@ -44,11 +45,26 @@ export default function ReferenceDataShow({
   name
 }: PropType): React.ReactElement {
   const cName: ResourceTypes = name as ResourceTypes
-
```
Comment (IanMayo) [inline src/resources/reference-data/ReferenceDataShow.tsx:47]: Hello @TahaKhanAbdalli - I don't like to see this array of titles twice in the code.  Could you please just declare it once?

Comment (TahaKhanAbdalli) [reply src/resources/reference-data/ReferenceDataShow.tsx:47]: Sorry, are you talking about this code?
image

Comment (IanMayo) [reply src/resources/reference-data/ReferenceDataShow.tsx:47]: Yes - it's present in two places.

**Review by IanMayo — APPROVED** [review]
Tested and working

**Pull request merged**